greenwave/policies.py

```diff
@@ -512,23 +486,61 @@ def summarize_answers(answers):
     Returns:
         str: Human-readable summary.
     """
-    test_answers = [
-        answer for answer in answers
-        if answer.is_test_result
-    ]
-    if not test_answers:
-        return 'no tests are required'
-
-    summary = _summarize_answers_without_errored(test_answers)
-
-    errored_count = len([
-        answer for answer in test_answers
-        if isinstance(answer, TestResultErrored)
-    ])
-    if errored_count:
-        summary += f' ({errored_count} {"error" if errored_count == 1 else "errors"})'
-
-    return summary
+    # let's count! counting is fun!
+    test_count = 0
+    failed_nontest_count = 0
+    failed_test_count = 0
+    errored_test_count = 0
+    incomplete_test_count = 0
+    missing_test_count = 0
+    for answer in answers:
```

Comment:
Instead of using `isinstance()` and having to extend the branches below when adding new Answer subclass, it would be better to wrap the counters in an object and let each answer object update it:

```python
summary = Summary()
for answer in answers:
    answer.summarize(summary)
return summary.as_text()
```